app/jobs/cleanup_account_job.rb

```diff
@@ -1,18 +1,32 @@
 class CleanupAccountJob < ActiveJob::Base
   def perform(account)
-    cleanup_solr(account)
-    cleanup_fedora(account)
+    # Ensure all account endpoints are active/configured
+    account.switch do
+      cleanup_solr(account)
+      cleanup_fedora(account)
+      # Redis doesn't need 'account', as switch changes Redis::Namespace to correct namespace
+      cleanup_redis
```

Comment:
No, my point is that those two jobs are designed to work in such a way that they still can be used when an account is broken.  `account.switch!` will fail if, e.g, `solr_endpoint_id` is nil, but we still want as many subtasks as possible to at least spawn, if not succeed.  I am trying to preserve that durability and idempotence.  

While correct for the general case, I believe you are mistaken about `switch` behavior **for these particular jobs**, since they take the account as an argument.  They don't use the switch-configured settings, they use the argument.  This is similar to the account *creation* jobs, another place where `switch!` cannot be called (yet).